addons/website_event/static/src/client_action/website_event_ticket_registration_dialog.js

```diff
@@ -0,0 +1,144 @@
+    static template = "website_event.WebsiteEventTicketRegistration";
+    static components = { Dialog };
+    static props = {
+        close: Function,
+        data: Object,
+    };
+
+    setup() {
+        this.orm = useService("orm");
+        this.registrationCount = 0;
+        this.isEventFull = useState({ value: false });
+        this.registrations = useState(this.buildRegistrations());
+        this.registrationsToDelete = useState([]);
+    };
+
+    get data() {
+        return this.props.data;
+    };
+
+    buildRegistrations(){
+        let registrations = [];
+        if (this.props.data.registrations) {
+            this.props.data.registrations.forEach(registration => {
+                this.registrationCount++;
+                registrations.push({
+                    index: this.registrationCount,
+                    registrationId: registration.registration_id,
+                    ticketTypeId: registration.ticket_id,
+                    ticketTypeName: registration.ticket_name,
+                    answers: registration.answers,
+                });
+                this.updateTicketSeatsAvailability(registration.ticket_id, undefined, 1);
+            });
+            this.updateEventSeatsAvailability(undefined, this.props.data.registrations.length);
+        }
+        else {
+            this.props.data.tickets.forEach(ticketType => {
+                for (let i = 1; i <= ticketType.quantity; i++) {
+                    this.registrationCount++;
+                    registrations.push({
+                        index: this.registrationCount,
+                        ticketTypeId: ticketType.id,
+                        ticketTypeName: ticketType.name,
+                        answers: Object.assign({}, ...this.props.data.event.specific_question_ids.map((question) => (
+                            { [question.id]: (this.registrationCount === 1 && this.props.data.default_first_attendee !== undefined) ? this.props.data.default_first_attendee[question.question_type] : '' }
+                        ))),
+                    })
+                }
+                this.updateTicketSeatsAvailability(ticketType.id, undefined, ticketType.quantity);
+                this.updateEventSeatsAvailability(undefined, ticketType.quantity);
+            });
+        }
+        return registrations;
+    }
+
+    getAvailableTicketTypes(actualTicketTypeId) {
+        return this.props.data.event.tickets_ids.filter((ticket) => ticket.id !== actualTicketTypeId);
+    };
+
+    updateTicketSeatsAvailability(ticket_id, qtyAdded = 0, qtyRemoved = 0) {
+        if (this.props.data.event.tickets_ids.length === 0) return;
+        const ticketIndex = this.props.data.event.tickets_ids.findIndex((ticket) => ticket.id === ticket_id);
+        this.props.data.event.tickets_ids[ticketIndex].seats_available += qtyAdded - qtyRemoved;
+    };
+
+    updateEventSeatsAvailability(qtyAdded = 0, qtyRemoved = 0) {
```

Comment:
When I proposed this I didn't think about the fact that we can do like in python. So instead, we could do:
`...(qtyParams = {})` and when calling the method just do `...({ qtyAdded: 1 })`. And same for the function above.
What do you think ?

Reply:
what would be the benefit of this dict ?
I mean how am I going to do this :
`this.data.event.seats_available += qtyAdded - qtyRemoved;`
If I don't know which value is or is not in the dict ?
I can check the keys but it's going to complexify everything no ?

Wasn't is simpler to have a parameter that represent the operation (positive or negative) as we will never use both at the same time ? (`quantityOperation` maybe ?)